src/main/java/com/fasterxml/jackson/databind/util/ExceptionUtil.java

```diff
@@ -0,0 +1,48 @@
+package com.fasterxml.jackson.databind.util;
+
+/**
+ * Utilitity methods for dealing with exceptions/throwables
+ *
+ * @since 2.15
+ */
+public class ExceptionUtil {
+    private ExceptionUtil() {}
+
+    /**
+     * It is important never to catch all <code>Throwable</code>s. Some like
+     * {@link InterruptedException} should be rethrown. Based on
+     * <a href="https://www.scala-lang.org/api/2.13.10/scala/util/control/NonFatal$.html">scala.util.control.NonFatal</a>.
+     *
+     * @param throwable to check
+     * @return whether the <code>Throwable</code> is a fatal error
+     */
+    public static boolean isFatal(Throwable throwable) {
+        return (throwable instanceof VirtualMachineError
+                || throwable instanceof ThreadDeath
+                || throwable instanceof InterruptedException
+                || throwable instanceof LinkageError);
```

Comment:
I am not sure `LinkageError` should be included in this list: isn't it one of failures that specifically can occur on various platforms trying to load optional extensions?

Reply:
I wouldn't be against removing LinkageError here or being specific. Some LinkageErrors seem more fatal than others - see the subclasses here - https://docs.oracle.com/javase/7/docs/api/java/lang/LinkageError.html

NoClassDefError seems least fatal but in use cases where I added this util to be used, we tend to have the Class already.

Reply:
Yeah the concern I have is specifically that the optional loading should not die on missing classes; that's pretty much the main use case. I wish actual test case. But conversely I haven't seen reports where catching one of fatal errors here have been problematic -- theoretically I could see `InterruptedException` catching being problematic, of these I think.